examples/iterate_over_table_info.py

```diff
@@ -0,0 +1,78 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Run arbitrary operations across multiple tables
+# MAGIC
+
+# COMMAND ----------
+
+# TODO: update the table mathching pattern:
+from_tables = "discoverx_sample.*.*"
+
+# COMMAND ----------
+
+from discoverx import DX
+
+dx = DX()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Define a custom function
+
+# COMMAND ----------
+
+from discoverx.table_info import TableInfo
+
+def my_function(table_info: TableInfo):
+
+  # TODO: Add your logic here
+  print(f"{table_info.catalog}.{table_info.schema}.{table_info.table} has {len(table_info.columns)} columns")
+  
+  if table_info.tags:
+    print(f" - catalog_tags: {table_info.tags.catalog_tags}")
+    print(f" - schema_tags: {table_info.tags.schema_tags}")
+    print(f" - table_tags: {table_info.tags.table_tags}")
+    print(f" - column_tags: {table_info.tags.column_tags}")
+
+  # The returned objects will be concatenated in the map result
+  return table_info
+  
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Run the custom fuctions for multiple tables
+
+# COMMAND ----------
+
+result = (dx
```

Comment:
Could we add an example that corresponds to the other existing examples, just adding the tag-functionality? I.e. an examples that does `dx.from_tables().with_tags().with_sql().apply()` ...

Reply:
I was planning to do a separate PR for the example and a new function that filters by tag.

Reply:
Right now the only function that can leverage tags is the "map".

Reply:
Will the above, i.e. using something else than map, fail with a meaningful error message in that case?

Reply:
It would not fail at all for other functions because `with_tags` only adds extra metadata to the table_info object, and it does not break any other functionality